analyses/oncoprint-landscape/util/oncoplot-functions.R

```diff
@@ -0,0 +1,103 @@
+# This script defines custom functions to be sourced in the
+# `01-plot-oncoprint.R` script of this module.
+#
+# Chante Bethell for CCDL 2020
+#
+# # #### USAGE
+# This script is intended to be sourced in the script as follows:
+#
+# source(file.path("util", "oncoplot-functions.R"))
+
+prepare_and_plot_oncoprint <- function(maf_df,
+                                       cnv_df,
+                                       metadata,
+                                       fusion_df = NULL,
+                                       goi_list = NULL,
+                                       color_palette,
+                                       running_in_figures_script = FALSE) {
+  # Given maf, cnv, and fusion data.frames prepared in `00-map-to-sample_id.R`,
+  # along with a list of genes prepared in the `interaction-plots` directory,
+  # plot an oncoprint landscape figure.
+  #
+  # Args:
+  #   maf_df: data.frame with data from a MAF file
+  #   cnv_df: data.frame with copy number variant data
+  #   metadata: data.frame with the relavant metadata
+  #   fusion_df: data.frame with fusion data. This is NULL by default.
+  #   goi_list: a list or vector of genes of interest that should be
+  #             represented on the oncoprint. This is NULL by default.
+  #   color_palette: vector of colors (hex codes) corresponding to the data
+  #                  that will be represented on the oncoprint
+  #   running_in_figures_script: TRUE or FALSE statement indicating whether
+  #                              or not this function is being run in the
+  #                              figures generation script
+  
+  if (!is.null(fusion_df)) {
+    # Bind rows of maf and fusion data frames
+    maf_df <- dplyr::bind_rows(maf_df, fusion_df)
+  }
+  
+  if (!is.null(goi_list)) {
+    # Filter `goi_list` to include only the unique genes of interest
+    # that are also in `maf_df`
+    goi_list <- unique(goi_list[goi_list %in% maf_df$Hugo_Symbol])
+  }
+  
+  # Convert into MAF object
+  maf_object <-
+    read.maf(
+      maf = maf_df,
+      clinicalData = metadata,
+      cnTable = cnv_df,
+      removeDuplicatedVariants = FALSE,
+      vc_nonSyn = c(
+        "Frame_Shift_Del",
+        "Frame_Shift_Ins",
+        "Splice_Site",
+        "Nonsense_Mutation",
+        "Nonstop_Mutation",
+        "In_Frame_Del",
+        "In_Frame_Ins",
+        "Missense_Mutation",
+        "Fusion",
+        "Multi_Hit",
+        "Multi_Hit_Fusion",
+        "Hom_Deletion",
+        "Hem_Deletion",
+        "amplification",
+        "gain",
+        "loss"
+      )
+    )
+  
+  # If this function is being run in the figures generation script, return the
+  # maf object (instead of the final plot)
+  if (running_in_figures_script) {
```

Comment:
Can you explain this distinction? Why are you plotting it in one situation and not in the other? Is there a way we can revise `analyses/oncoprint-landscape/01-plot-oncoprint.R` so we have less of a branch off? Secondly, I would have expected this function to always plot based on the name `prepare_and_plot_oncoprint`.

Reply:
The issue here is that the `coOncoplot` function (that I used to lay out the two specific oncoprints side by side in the figure generation script expects two maf objects, while the `oncoplot` function used to produce each individual oncoprint in the analyses/oncoprint-landscape/plots module only expects one maf object.

Should I maybe add the coOncoplot step to this function for `running_in_figures_script` = `TRUE` and add the function argument `maf_df_2` = `NULL` for the purpose of consistency?

Reply:
Do you end up using `coOncoplot` in both scenarios? If yes, then, let's find a way to incorporate it into the `prepare_and_plot_oncoprint ` function, if no, then we might want to remove it from this particular function.

Reply:
No, should I should remove the `oncoplot` step from this function and put it in the `01-plot-oncoprint.R` script in that case?

Reply:
Upon digging into the code a bit I would say, yes. Put the `oncoprint` step into `01-plot-oncoprint.R` and remove it from `prepare_and_plot_oncoprint`. Then you will should probably: 

1) Get rid of the `running_in_figures_script` argument. 
2) Rename `prepare_and_plot_oncoprint` it something that indicates it won't do the plotting, but is preparing it for the plotting. `prepare_for_oncoprint` or something like that.